x-pack/plugins/cases/server/cases_analytics_index/index.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ConcreteTaskInstance,
+  RunContext,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import type { CoreSetup, ElasticsearchClient, ISavedObjectsRepository } from '@kbn/core/server';
+import {
+  CASE_COMMENT_SAVED_OBJECT,
+  CASE_SAVED_OBJECT,
+  CASE_USER_ACTION_SAVED_OBJECT,
+} from '../../common/constants';
+import { TASK_NAME } from './constants';
+import { CaseAnalyticsIndexSyncTaskFactory } from './task_factory';
+import type { CasesServerStartDependencies } from '../types';
+
+export const registerCaseAnalyticsIndexSyncTask = ({
+  core,
+  taskManager,
+}: {
+  core: CoreSetup<CasesServerStartDependencies>;
+  taskManager: TaskManagerSetupContract;
+}) => {
+  const getSavedOjectClient = async (): Promise<ISavedObjectsRepository> => {
+    const [{ savedObjects }] = await core.getStartServices();
+    return savedObjects.createInternalRepository([
```

Comment:
What is the purpose of `createInternalRepository`?

The documentation says:
> Creates a {@link ISavedObjectsRepository | Saved Objects repository} that uses the internal Kibana user for authenticating with Elasticsearch.

and `ISavedObjectsRepository` is 
> The savedObjects repository contract.

but what does this actually mean in practice?